LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file and sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its LocalFolder directory.
+* Support web apps not installed and managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+* Introduce a non-standard Web API
+
+## Proposed Solution 
+
+### Existing behavior
+
+`navigator.storage.getDirectory()` is an existing API that returns a promise to a FileSystemDirectoryHandle representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
+
+### New behavior
+
+Within a Microsoft Store PWA context, the directory handle for OPFS root will contain an additional entry that represents the app's LocalFolder location. This entry can be retrieved as a `FileSystemDirectoryHandle` by calling `.entries()` or `.getDirectory(...)` on the OPFS root directory handle.
+
+The LocalFolder directory handle and its contents will have properties different from other OPFS directory handles - they will only have read and delete permissions instead of read and write permissions.
+
+```JS
+    // .getDirectory()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+    }
+```
+
+```JS
+    // .entries()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        for await (const [key, value] of opfsRoot.entries()) {
+            if (key == `microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt`) {
+                // ...
+            }
+        }
+    }
+```
+
+### Deleting files
+
+LocalStorage contents can be cleared by calling `.removeEntry(...)` from the OPFS root directory handle.
+```JS
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        await opfsRoot.removeEntry("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", { recursive: true }); 
+    }
+```
+
+Alternatively, contents in LocalFolder can be deleted by calling `remove()` on file and directory handles directory.
+
+```JS
+    let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+
+    for await (const [key, value] of localStorage.entries()) {
+        value.remove();
+    }
+```
+
+### Entry name
```

Comment:
We should reiterate in this section that:

1. the entry will only exist for PWA's installed through the Microsoft Store, and even then, only when the OPFS API is being used within the context of the PWA.
2. that it won't be accessible through background web workers
3. that it will exist for Microsoft Store PWA's, even if the LocalFolder itself is empty.
4. that the entry can't be deleted, but that its internal contents _can_ be deleted

Reply:
1 and 2 are stated in the "Microsoft Store PWA context" section. 

Re. 3. I think it's ok for the entry to not exist for MS Store PWAs if for some reason the directory is not found on the file system. 

4. I'm reconsidering this requirement.

Reply:
Updated deletion behavior.